### PR TITLE
19 create data model for markers

### PR DESCRIPTION
**Pull request by sooble13**
## Description
This PR introduces a new Marker schema, which includes information as longitude, latitude, timestamps, and marker types.

## Related Issue
Closes https://github.com/UNLV-CS472-672/2024-S-GROUP1-Roadwatch/issues/19

## Type of Changes
- [x] longitude
- [x] latitude
- [x] timestamp
- [x] marker type
      
## Things to Update
- [ ] Might need to add more fields.
- [ ] Accident Field
- [ ] Other Field
- [ ] Traffic Field
- [ ] Construction Field

## Additional Notes
This is a simple Marker Schema. Most likely, there are more fields that will need to be added in the future.

### CODE REVIEW

File: server/src/models/Marker.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        type: Number,
+        required: true,
+    },
+    timestamp: {
+        type: Date,
+        default: Date.now,
+    },
+    type: {
+        type: String,
+        enum: ['default', 'pothole', 'sbump', 'closure','xwalk'], // Add more types as needed
```
Comment (Aligary) [inline server/src/models/Marker.ts:19]: These are useful types of markers for now. We can look into adding a car accident type or 'suggestion' type in the future. Not sure if speed bumps or crosswalks are needed.

**Review by Aligary — APPROVED** [review]
Thanks for your work on this! It looks good to me

File: server/src/models/Marker.ts (hunk 1 above)
Comment (sooble13) [reply server/src/models/Marker.ts:19]: Great idea! I was having a bit of trouble thinking of some basic types, but accidents and suggestions sound good to me. Perhaps maybe other would suffice as well? Could update this in the future anyways. I'll add your suggestions to the "Things to Update" portion.

**Comment by Linkaloo** [comment]
Same thing as your other pr you have a commit behind with the instruction page and a commit ahead without it. which may be causing your weird merge issues. If merging main works on the other branch then also do it here.

**Review by Linkaloo — APPROVED** [review]
Looks good

**Review by trevortippery — APPROVED** [review]
Thank you for fixing the branch issues. Looks good now. Approved 🎉

**Pull request merged**